code/ATMOSPHERICS/components/trinary_devices/pressure_valve.dm

```diff
@@ -76,40 +136,74 @@
 
 // NPCs and assorted gremlins
 /obj/machinery/atmospherics/trinary/pressure_valve/npc_tamper_act(mob/living/L)
-	if (prob(33))
-		open_on_above_threshold = !open_on_above_threshold
-		investigation_log(I_ATMOS,"was switched to open [(open_on_above_threshold ? "above" : "below")] [pressure_threshold]kPa by [key_name(L)]")
-
-	else if (prob(33))
-		pressure_threshold = rand(0, 9000)
-		investigation_log(I_ATMOS,"had its threshold configured to open [(open_on_above_threshold ? "above" : "below")] [pressure_threshold]kPa by [key_name(L)]")
-
-	else
-		toggle_enabled()
-		investigation_log(I_ATMOS,"was [(enabled ? "enabled" : "disabled")] by [key_name(L)]")
+	switch(rand(0, 5))
+		if(0)
+			toggle_enabled()
+			investigation_log(I_ATMOS,"was [(enabled ? "enabled" : "disabled")] by [key_name(L)]")
+		if(1)
+			open_on_above_threshold = !open_on_above_threshold
+			investigation_log(I_ATMOS,"was switched to open if controls is [(open_on_above_threshold ? "ABOVE" : "BELOW")] the threshold by [key_name(L)]")
+		if(2)
+			mode = pick(CVALVE_MODE_PRESSURE, CVALVE_MODE_TEMPERATURE)
+			investigation_log(I_ATMOS,"had it's compare mode set to [mode] by [key_name(L)]")
+		if(3)
+			mode = pick(CVALVE_INPUT_LEFT, CVALVE_INPUT_CENTER, CVALVE_INPUT_RIGHT)
+			investigation_log(I_ATMOS,"had it's control pipe set to [control_input] by [key_name(L)]")
+		if(4)
+			mode = pick(CVALVE_THRESHOLD_LEFT, CVALVE_THRESHOLD_CONSTANT, CVALVE_THRESHOLD_RIGHT)
+			investigation_log(I_ATMOS,"had it's threshold source set to [threshold_source] by [key_name(L)]")
+		if(5)
+			constant_threshold = rand(0, 9000)
+			investigation_log(I_ATMOS,"had it's threshold constant set to [constant_threshold] by [key_name(L)]")
+	investigation_log(I_ATMOS,"current configuration: open if [control_input] [mode] [(open_on_above_threshold ? "ABOVE" : "BELOW")] [threshold_source][threshold_source == CVALVE_THRESHOLD_CONSTANT ? "[constant_threshold][mode2unit[mode]]" : ""]")
 
 	update_icon()
 
 // --- Behaviour ---
-
 /obj/machinery/atmospherics/trinary/pressure_valve/process()
 	. = ..()
 	if (!enabled)
 		close()
 		return
 
-	if (open_on_above_threshold && (air2.pressure > pressure_threshold) || !open_on_above_threshold && (air2.pressure < pressure_threshold))
+	var/input = 0
+	var/threshold = 0
+	switch(control_input)
+		if (CVALVE_INPUT_LEFT) input = get_valve_mode_gas_value(pipe_flags & IS_MIRROR ? air3 : air1)
```

Comment:
Formatting, we don't do one-line `if`s, this is bad for reading; especially with the weird ternary operator after